src/test/kfs_rest_parser_test.cpp

```diff
@@ -297,6 +297,31 @@ TEST_F(KFSRestParserTest, parseValidRequestBOOL) {
     ASSERT_THAT(proto.inputs()[0].contents().bool_contents(), ElementsAre(true, true, false, false));
 }
 
+TEST_F(KFSRestParserTest, parseRequestWithZeroDim) {
+    std::string request = R"({
+    "inputs" : [
+        {
+        "name" : "input0",
+        "shape" : [ 10, 224, 0, 3 ],
+        "datatype" : "INT64",
+        "data" : [ ]
+        }
+    ]
+    })";
+    auto status = parser.parse(request.c_str());
+    ASSERT_EQ(status, StatusCode::OK);
+
+    auto proto = parser.getProto();
+    ASSERT_EQ(proto.inputs_size(), 1);
+    ASSERT_EQ(proto.inputs()[0].name(), "input0");
+    ASSERT_THAT(proto.inputs()[0].shape(), ElementsAre(10, 224, 0, 3));
+    ASSERT_EQ(proto.inputs()[0].datatype(), "INT64");
+    ASSERT_EQ(proto.inputs()[0].contents().fp32_contents_size(), 0);
```

Comment:
We should check _content_size for every datatype here. Please add comment with explanation why checking others than int64_contents_size makes sense